libkbfs/folder_block_ops.go

```diff
@@ -2585,14 +2575,36 @@ func (fbo *folderBlockOps) getDeferredWriteCountForTest(lState *lockState) int {
 	return len(fbo.deferredWrites)
 }
 
+func (fbo *folderBlockOps) updatePointer(kmd KeyMetadata, oldPtr BlockPointer, newPtr BlockPointer) {
+	updated := fbo.nodeCache.UpdatePointer(oldPtr.Ref(), newPtr)
+	if !updated {
+		return
+	}
+	// Prefetch the new reference, but only if it already exists in the block
```

Comment:
Oh, one more reason, actually: the old ref tells us the type of the block, hence allows us to call `block.NewEmpty()`. Not sure of a good way to get that information from the ref.